parse-zoneinfo/src/line.rs

```diff
@@ -877,6 +859,190 @@ pub struct Rule<'a> {
     pub letters: Option<&'a str>,
 }
 
+impl<'a> Rule<'a> {
+    fn from_str(input: &'a str) -> Result<Self, Error> {
```

Comment:
Unfortunately it appears that `FromStr` doesn't support capturing a lifetime, so I just made this an inherent method also called `from_str()` instead.